### PR TITLE
Create initial set of contribution docs

### LINKED ISSUE #44: Clarify instructions for setting up local development environment
This is still not obvious.

### PR DESCRIPTION
**Pull request by alysivji**
Closes #44 and #81 

Building off the work of @uncojohnco in #83, I created contribution docs using an environment variables file that is read into Docker-Compose automatically. Database credentials are hardcoded in the `docker-compose.yaml`; this is by far the cleanest solution I could come up with.

Johnny, you wanna give the new instructions a shot?

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -21,20 +21,12 @@ services:
       context: .
       dockerfile: ./docker/dev/Dockerfile
     command: ["gunicorn", "busy_beaver:app", "-b", "0.0.0.0:5100", "--reload", "--timeout",  "100000"]
+    env_file: .env
     environment:
```
Comment (alysivji) [inline docker-compose.yml]: I will be using YAML anchoring for the task queue containers (which require the same vars) so we don't have to keep repeating this

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Development Environment
+
+It is recommended that users create a personal Slack workspace to use for bot development. This will allow for independent development without having to wait for project maintainers to grant access to the Busy-Beaver development Slack.
+
+### Slack Integration
+
+1. [Create a Slack workspace](https://get.slack.help/hc/en-us/articles/206845317-Create-a-Slack-workspace)
+1. [Create a Slack App](https://api.slack.com/apps) and set the development workspace to the workspace from the previous step
+1. Configure Slack app settings (defined below)
```
Comment (uncojohnco) [inline CONTRIBUTING.md]: Maybe have the `add bot user` step first as I couldn't install an app without having a feature defined?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+1. Install application in workspace
+1. Locate the `Bot User OAuth Access Token`, we will need that later on
+
+#### Slack Settings
+
+- Add a bot user
```
Comment (uncojohnco) [inline CONTRIBUTING.md]: Some screenshots to help guide contributors
add-bot-user-1
add-bot-user-2

**Comment by uncojohnco** [comment]
I had a quick this evening and looks good so far. :+1: 

I think I may have a few suggestions for improvements, so it might be easier to merge this into master? Then I can open up a PR against the contribution.md with suggested changes?

File: CONTRIBUTING.md (hunk 2 above)
Comment (alysivji) [reply CONTRIBUTING.md]: I was thinking about linking, but each app has a different URL.

File: CONTRIBUTING.md (hunk 3 above)
Comment (alysivji) [reply CONTRIBUTING.md]: That's awesome!

**Pull request merged**